homeassistant/components/tplink/__init__.py

```diff
@@ -115,9 +187,16 @@ async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     if unload_ok := await hass.config_entries.async_unload_platforms(entry, PLATFORMS):
         hass_data.pop(entry.entry_id)
     await device.protocol.close()
+
     return unload_ok
 
 
+async def async_remove_entry(hass: HomeAssistant, entry: ConfigEntry) -> None:
+    """Cleanup caches before removing config entry."""
+    if len(hass.config_entries.async_entries(DOMAIN)) == 1:
```

Comment:
```suggestion
    if hass.config_entries.async_entries(DOMAIN):
```
Maybe? If this only checks for the list emptiness.

Reply:
Method now dropped as we're not using the central `Store`